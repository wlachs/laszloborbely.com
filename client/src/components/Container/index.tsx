import React, {type ReactNode} from 'react';

function Container(props: {
	children?: React.JSX.Element | ReactNode | string;
}) {
	const {children} = props;
	return (
		<div className='d-flex flex-column align-items-center h-100 w-100 flex-grow-1'>
			{children}
		</div>
	);
}

export default Container;
