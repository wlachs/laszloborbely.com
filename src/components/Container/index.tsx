import { type ReactNode } from 'react';

function Container(props: {
	children?: React.JSX.Element | ReactNode | string;
}) {
	const { children } = props;
	return (
		<div className='d-flex flex-column align-items-center h-100 col-11 col-lg-12 m-auto flex-grow-1'>
			{children}
		</div>
	);
}

export default Container;
