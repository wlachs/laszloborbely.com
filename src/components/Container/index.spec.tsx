import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, it } from 'vitest';

import Container from './index.tsx';

describe('Container component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should always render child elements', () => {
		const child = 'child';
		const { container } = render(<Container>{child}</Container>);
		expect(screen.getByText(child)).toBeDefined();
		expect(container).toMatchSnapshot();
	});
});
