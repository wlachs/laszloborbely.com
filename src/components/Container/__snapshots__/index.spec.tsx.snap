// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container component > should always render child elements 1`] = `
<div>
  <div
    class="m-auto flex flex-col items-center"
  >
    child
  </div>
</div>
`;
